api/envoy/extensions/filters/http/grpc_field_extraction/v3/config.proto

```diff
@@ -0,0 +1,131 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.grpc_field_extraction.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_field_extraction.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_field_extraction/v3;grpc_field_extractionv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: gRPC Field Extraction]
+// gRPC Field Extraction :ref:`configuration overview
+// <config_http_filters_grpc_field_extraction>`.
+// [#extension: envoy.filters.http.grpc_field_extraction]
+
+// GrpcFieldExtractionConfig supports extracting the fields from first gRPC
+// request message no matter if it is unary or streaming and inject the result
+// as a header. It won't propagate headers until the extraction and the
+// injection done so the following filters can use it. No op if isn't a gRPC
+// method or its type cannot be found in the proto descriptor.
+//
+// The target field should be singular primitive type or repeated primitive
+// type and its value will be extracted in string format.
+//
+//
+// The output is a serialized google.protobuf.Struct of field<StringValue> -> extract values<ListValue of StringValue>. It will be injected in the `header` with base64 encoded format.
+//
+// For example, we have the following request definition for the gRPC method `/pkg.svc/Method`.
+//
+// message MethodRequest {
+//   string method_foo = 1;
+//   string method_bar = 2;
+//   repeated int32 method baz = 3;
+//   ...
+// }
+//
+// This is the filter config(expressed in JSON format).
+// {
+//   "proto_descriptor_bin":{...},
+//   "extractions_by_method": {
+//      "/pkg.svc/Method":{
+//         "request_field_extractions":{
+//            "name_foo":{
+//               "header":"grpc-field-extraction-output1-bin"
+//            },
+//            "name_bar":{
+//               "header":"grpc-field-extraction-output1-bin"
+//            }
+//            "name_baz":{
+//               "header":"grpc-field-extraction-output2-bin"
+//            }
+//         }
+//      }
+//   },
+//   ...
+// }
+//
+//
+// During runtime, the filter receives the following `MethodRequest` message(expressed in JSON format).
+// {
+//   name_foo: "val_foo",
+//   name_bar: "val_bar",
+//   name_baz: [123, 456]
+// }
+//
+// The filter will injects the following 2 request headers(expressed in JSON format).
+//
+// rpc-field-extraction-output1-bin: {
```

Comment:
It's not clear to me the reason for combining multiple fields under one header. Can you provide the use case for it?

Reply:
TBH, I don't have a practical use case either currently. In our case, we are doing RBAC relying on extracting the request field and calling some extAuthz server. Maybe it could be possible that there are multiple RBACs required so different destination headers help decouple.

This API design doesn't want to block this kind of possibility?

Reply:
Unless there is a good use case, it's better to keep it focused. I'd recommend to keep it simple but structure in a way that allows extensibility.

Reply:
Updated to use a default dynamic metadata for now. PTAL.